grafana-plugin/frontend_guidelines.md

```diff
@@ -0,0 +1,112 @@
+# OnCall Frontend Guidelines
+
+## Ways of working
+
+### Code reviews
+
+- PR can be merged when the following conditions are met:
+   pipeline succeeded without bypassing checks
+  ere is at least 1 approval from the frontend team and no opened remarks from reviewers other than the approver
+  l comments are replied, answered, addressed or discussed separately
+   make use of builtin GH PR reviews
+  nor comments that don’t need to be addressed right away and are not a blocker for a merge are marked with “Minor: “ prefix
+  nless there is some important hotfix to make and no frontend mates are available*
+
+## Technical conventions
+
+### Type-safety
+
+- Don’t use implicit / explicit `any` or @ts-ignore
+
+### Use auto generated types for new features
+
+- For new features and endpoints that are covered by OpenApi schemas, we should use auto generated types and typed http client
+
+### Naming conventions on most used variables such as handlers, booleans etc
+
+- Use descriptive naming (such as the use of is) that follows natural convention such as isFormVerified
+or isAuthenticated instead of authenticated or other forms
+- Use descriptive naming for event handlers such as onIconClick or onInputChange rather than handleChange or other namings
+- Include the verb part in the functions’ names
+- For 3 or more function arguments use object destructuring
+<https://github.com/ryanmcdermott/clean-code-javascript?tab=readme-ov-file#function-arguments-2-or-fewer-ideally>
+- For functions that return other functions getBlaBlaHandler (in case of handlers) or getBlaBlaFn
+(if it’s not handler that is returned)
+- Don’t use higher-ordered functions without clear the need
+
+## State management
+
+### Don’t use returned values from MobX actions
+
+- MobX actions should not return value that is later used within components. Instead MobX actions should update
+observables and then components should consume observables to reflect data updates.
+
+### Leverage small global stores over local state passed as props over multiple levels
+
+- Create small specialized store in MobX even for small feature, e.g.:
+  alert_receive_channel_connected_channels.ts
+  alert_receive_channel_webhooks.ts
+- Instead of relying on local state that is passed down the React tree over multiple levels
+
+### Use global decorators / custom hooks / utilities consistently
+
+- For example:
+  ithGlobalNotification to set successful / failing notification based on MobX action’s result
+  utoLoadingState to manage loading statuses of async actions
+  eIsLoading to consume loading state
+  eDrawer to manage drawers
+  eConfirmModal to manage confirm modal
+  obal helpers
```

Comment:
seems odd here, looks like the first letter of every line got cut off?

Reply:
It was broken by copy pasting from markdown previewer tool, I fixed all the places now, thanks